packages/make-styles/src/makeStaticStyles.ts

```diff
@@ -0,0 +1,32 @@
+import { CAN_USE_CSS_VARIABLES } from './constants';
+import { createCSSVariablesProxy } from './runtime/index';
+import { resolveStaticStyleRules } from './runtime/resolveStaticStyleRules';
+import { MakeStaticStylesStyleRule, MakeStylesOptions, MakeStaticStylesStyleFunctionRule } from './types';
+
+/**
+ * Register static css.
+ * @param styles styles object or string.
+ */
+export function makeStaticStyles<Tokens>(styles: MakeStaticStylesStyleRule<Tokens>) {
+  const styleCache: Record<string, true> = {};
+
+  function computeClasses(options: MakeStylesOptions<Tokens>): void {
+    // TODO: remove proxy
+    const tokens = CAN_USE_CSS_VARIABLES ? createCSSVariablesProxy(options.tokens) : options.tokens;
+
+    const styleRules =
+      typeof styles === 'function' ? (styles as MakeStaticStylesStyleFunctionRule<Tokens>)(tokens) : styles;
+    const resolvedStyleRule = resolveStaticStyleRules(styleRules);
+
+    const cacheKey = options.renderer.id;
+    const cacheValue = styleCache[cacheKey];
+    if (cacheValue) {
+      return;
+    }
+
+    options.renderer.insertDefinitions(resolvedStyleRule, !!options.rtl);
+    styleCache[cacheKey] = true;
+  }
+
+  return computeClasses;
+}
```

Comment:
I was thinking about it and I don't think that we should support tokens for global styles 🤔 Even in it's definition we have a word "static"... This can simplify implementation much:

```suggestion
import { resolveStaticStyleRules } from './runtime/resolveStaticStyleRules';
import { MakeStylesOptions, MakeStaticStyles } from './types';

/**
 * Register static css.
 * @param styles styles object or string.
 */
export function makeStaticStyles(styles: MakeStaticStyles) {
  const styleCache: Record<string, true> = {};

  function computeClasses(options: Pick<MakeStylesOptions<never>, 'renderer'>): void {
    const resolvedStyles = resolveStaticStyleRules(styles);

    // Static styles should be registered once per renderer (document)
    if (styleCache[options.renderer.id]) {
      return;
    }

    options.renderer.insertDefinitions(resolvedStyles, false /* static rules do not support RTL transforms */);
    styleCache[options.renderer.id] = true;
  }

  return computeClasses;
}
```